src/Simulation/TargetDefinitions/Decompositions/ExpUtil.qs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Quantum.Intrinsic {
+    open Microsoft.Quantum.Diagnostics;
+
+    @EnableTestingViaName("Test.TargetDefinitions.ExpUtil")
+    internal operation ExpUtil (paulis : Pauli[], theta : Double, qubits : Qubit[], rotation : ((Pauli, Qubit) => Unit is Adj + Ctl)) : Unit is Ctl {
+        if (Length(paulis) != Length(qubits)) { fail "Arrays 'paulis' and 'qubits' must have the same length"; }
+        if (Length(paulis) == 1) {
+            rotation(paulis[0], qubits[0]);
+        }
+        else { // Length(paulis) > 1 
```

Comment:
Or `Length(paulis) == 0`? Does the `qubits[0]` reference below fail in that case?

Reply:
Thankfully no, because this utility is only called by `Exp` in ExpFromExpUtil and `ExpFrac` in ExpFracFromExpUtil with a check:
```qsharp
            if (Length(paulis) != 0) {
```
or
```qsharp
            let (newPaulis, newQubits) = RemovePauliI(paulis, qubits);

            if (Length(newPaulis) != 0) {
                ExpUtil(newPaulis, theta , newQubits, R(_, -2.0 * theta, _));
            }
            else {
                ApplyGlobalPhase(theta);
            }
```
Where `RemovePauliI` correctly handles empty lists.

Reply:
Maybe a quick comment mentioning this would be good, since it's potentially a gotcha if another place uses this operation without checking for length == 0?